```
components/transfer/index.tsx(164,65): error TS2345: Argument of type '(data: TransferItem) => boolean | undefined' is not assignable to parameter of type '(value: TransferItem, index: number, array: TransferItem[]) => boolean'.

  Type 'boolean | undefined' is not assignable to type 'boolean'.

    Type 'undefined' is not assignable to type 'boolean'.
```

typescript 语法有问题。

--- a/components/transfer/index.tsx
+++ b/components/transfer/index.tsx
@@ -161,7 +161,7 @@ export default class Transfer extends React.Component<TransferProps, any> {
     const { sourceSelectedKeys, targetSelectedKeys } = this.state;
     const moveKeys = direction === 'right' ? sourceSelectedKeys : targetSelectedKeys;
     // filter the disabled options
-    const newMoveKeys = moveKeys.filter(key => !dataSource.some(data => key === data.key && data.disabled));
+    const newMoveKeys = moveKeys.filter(key => !dataSource.some(data => key === !!(data.key && data.disabled)));
     // move items to target box
     const newTargetKeys = direction === 'right'
       ? newMoveKeys.concat(targetKeys)

--- a/components/transfer/index.tsx
+++ b/components/transfer/index.tsx
@@ -161,7 +161,7 @@ export default class Transfer extends React.Component<TransferProps, any> {
     const { sourceSelectedKeys, targetSelectedKeys } = this.state;
     const moveKeys = direction === 'right' ? sourceSelectedKeys : targetSelectedKeys;
     // filter the disabled options
-    const newMoveKeys = moveKeys.filter(key => !dataSource.some(data => key === data.key && data.disabled));
+    const newMoveKeys = moveKeys.filter(key => !dataSource.some(data => !!(key === data.key && data.disabled)));
     // move items to target box
     const newTargetKeys = direction === 'right'
       ? newMoveKeys.concat(targetKeys)

--- a/components/transfer/index.tsx
+++ b/components/transfer/index.tsx
@@ -161,7 +161,7 @@ export default class Transfer extends React.Component<TransferProps, any> {
     const { sourceSelectedKeys, targetSelectedKeys } = this.state;
     const moveKeys = direction === 'right' ? sourceSelectedKeys : targetSelectedKeys;
     // filter the disabled options
-    const newMoveKeys = moveKeys.filter(key => !dataSource.some(data => key === !!(data.key && data.disabled)));
+    const newMoveKeys = moveKeys.filter(key => !dataSource.some(data => !!(key === data.key && data.disabled)));
     // move items to target box
     const newTargetKeys = direction === 'right'
       ? newMoveKeys.concat(targetKeys)

